configurable_http_proxy/dbstore.py

```diff
@@ -0,0 +1,167 @@
+import json
+import logging
+import os
+from datetime import datetime
+
+from dataset import connect
+
+from configurable_http_proxy.store import BaseStore
+
+log = logging.getLogger(__name__)
+
+
+class DatabaseStore(BaseStore):
+    """A DBMS storage backend for configurable-http-proxy
+
+    This enables chp to run multiple times and serve routes from a central
+    DBMS. It uses SQLAlchemy as the database backend.
+
+    Usage:
+        Set the CHP_DATABASE_URL env var to any db URL supported by SQLAlchemy.
+        The default is "sqlite://chp.sqlite".
+
+        $ export CHP_DATABASE_URL="sqlite:///chp.sqlite"
+        $ configurable-http-proxy --storage-backend configurable_http_proxy.dbstore.DatabaseStore
+
+        Optionally you may set the table name by setting the CHP_DATABASE_TABLE.
+        The default is 'chp_routes'
+
+        $ export CHP_DATABASE_TABLE="chp_routes"
+
+    See Also:
+        * Valid URLs https://docs.sqlalchemy.org/en/14/core/engines.html#database-urls
+    """
+
+    default_db_url = "sqlite:///chp.sqlite"
+    default_db_table = "chp_routes"
+
+    def __init__(self):
+        super().__init__()
+        db_url = os.environ.get("CHP_DATABASE_URL", self.default_db_url)
```

Comment:
Instead of using environment variables ? Can we get the configs for this from the jupyter  config file ?
For example how the other jupyterhub proxy configs have it: https://jupyterhub-traefik-proxy.readthedocs.io/en/latest/toml.html#example-setup

Reply:
> I have merged the PR for github-actions - Could you rebase this PR based on that ? Also, could you help me understand why do you use `dataset` and not `sqlalchemy` directly here ?
> 
> `dataset` seems to need a bunch of other dependencies which we could avoid - https://github.com/pudo/dataset/blob/master/setup.py#L32

The rationale for `dataset` is that it reduces the effort to implement this feature by about 90%. To make this operationally useful, it needs to support db upgrades (migrations), so alembic is required anyway. banal is a very small price to pay for that.

Reply:
> Can we get the configs for this from the jupyter config file ?

I wouldn't mind, however as I'm not using chp in the context of jupyter,  jupyter config files are not useful to me.

Reply:
On the jupyter configs - makes sense
I guess I was more thinking of adding CLI arguments instead of env variables to be consistent with the other configs we use in configurable-http-proxy